### ISSUE TITLE
[Bug?] An empty file is determined to be binary.

### ISSUE THREAD

**Issue by uga-rosa** [ibhagwan/fzf-lua#567]
### Info

- Operating System: Ubuntu 20.04 on WSL2
- Shell: zsh
- Terminal: Windows Terminal
- `nvim --version`: NVIM v0.9.0-dev-6e8ed5aba
- `fzf --version`: 0.32.0

- [x] The issue is reproducible with `mini.sh`

<details>
<summary>Step to reproduce</summary>

```sh
$ cd /tmp
$ git clone git@github.com:ibhagwan/fzf-lua.git --depth 1
$ cat <<EOS > mini.vim
set hidden
set noswapfile

set rtp+=./fzf-lua
EOS
$ touch blank
$ nvim -u mini.vim
```

In nvim, use `:FzfLua files` to see the preview of `blank`.

</details>

### Description 

It is not a problem, so it is not a priority, but I feel uncomfortable with it, so I am reporting it as an issue.
As shown in the title, an empty file is determined to be a binary file in the preview.

**Comment by ibhagwan** [comment]
Thanks for reporting, can you check if the utility function works as expected?
>the second line should return false
```lua
:lua print(require"fzf-lua.utils".perl_file_is_binary("/bin/sh"))
:lua print(require"fzf-lua.utils".perl_file_is_binary("<empty file path>"))
```

**Comment by uga-rosa** [comment]
Unfortunately, both of these would be true. This function seems to be the cause.

**Comment by ibhagwan** [comment]
The fin room uses `perl -B` flag to test for binary, can you run the following in the shell?
```sh
perl -E 'exit((-B $ARGV[0])?0:1)' <empty file>
echo $? # should return 1
perl -E 'exit((-B $ARGV[0])?0:1)' /bin/sh
echo $? # should return 0

**Comment by uga-rosa** [comment]
This is also the same for both, which is 0.

**Comment by ibhagwan** [comment]
> This is also the same for both, which is 0.

That’s wrong result by perl then, not much I can do about it, what’s special about this file?
Does this happen with other empty files? Perhaps a WSL issue?

What does `file --dereference --mime <file>` return?

**Comment by ibhagwan** [comment]
Btw, this used to be done via `file --dereference --mime` (instead of `perl`) but it was way more inconsistent so I moved to `perl`.

**Comment by uga-rosa** [comment]
That seems to be the specification of -B in perl.  Would it be bad to also include a determination of whether or not the file is empty in this utility function?

**Comment by ibhagwan** [comment]
> That seems to be the specification of -B in perl. Would it be bad to also include a determination of whether or not the file is empty in this utility function?

https://github.com/ibhagwan/fzf-lua/commit/74072cc855aeeafa7f963ad75b73828e71132bff - free check since we already have the `stat` result of the file.
